test/cppsuite/src/main/database_operation.cpp

```diff
@@ -385,13 +385,15 @@ database_operation::update_operation(thread_worker *tc)
         /* Get the cursor associated with the collection. */
         scoped_cursor &cursor = cursors[coll.id];
 
-        /* Choose a random key to update. */
-        testutil_assert(coll.get_key_count() != 0);
-        auto key_id =
-          random_generator::instance().generate_integer<uint64_t>(0, coll.get_key_count() - 1);
-        auto key = tc->pad_string(std::to_string(key_id), tc->key_size);
+        /* Choose a random key to update using a random cursor. */
+        scoped_cursor rnd_cursor = tc->session.open_scoped_cursor(coll.name, "next_random=true");
+        int ret = rnd_cursor->next(rnd_cursor.get());
+        if (ret != 0)
+            testutil_die(ret, "Random cursor failed to find a key");
```

Comment:
I think we could end up with empty tables if we remove too much stuff. As long as it is `WT_NOTFOUND`, it is fine, we could add a trace for this. If this is a different error, we should probably stop the program.